src/bp/core/services/telemetry/hooks.ts

```diff
@@ -13,25 +13,60 @@ import { GhostService } from '..'
 import { BotService } from '../bot-service'
 import { JobService } from '../job-service'
 
-import { REDACTED, TelemetryStats } from './telemetry-stats'
+import { TelemetryStats } from './telemetry-stats'
 
-interface BotHooks {
+export interface BotHooks {
   botId: string
   hooks: Hook[]
 }
 
-interface Hook {
+export interface Hook {
   name: string
   type: string
   enabled: boolean
   lifecycle: string
+  path?: string
 }
 
-interface HookPayload {
+export interface HookPayload {
   perBots: BotHooks[]
   global: Hook[]
 }
 
+export async function getHooksLifecycle(botService: BotService, ghostService: GhostService): Promise<HookPayload> {
+  const botIds = await botService.getBotsIds()
+  const perBots = await Promise.map(botIds, async id => {
+    const botHooksPaths = await ghostService.forBot(id).directoryListing('/hooks', '*.js')
+    const lifecycles = parsePaths(botHooksPaths)
+    return { botId: calculateHash(id), hooks: lifecycles }
+  })
+  const globalHooksPaths = await ghostService.global().directoryListing('/hooks', '*.js')
+  const global = parsePaths(globalHooksPaths)
+
+  return { global, perBots }
+}
+
+function parsePaths(paths: string[]): Hook[] {
+  return paths.reduce((acc, curr) => {
+    const path = curr.split('/')
+    const lifecycle = path.shift() || ''
+    const hookName = path.pop()
+    const module = path.pop()
+    const isBuiltIn = !!(module && BUILTIN_MODULES.includes(module))
+    const [name, enabled] = parseHookName(hookName || '', isBuiltIn)
+
+    return [...acc, { name, lifecycle, enabled, type: isBuiltIn ? 'built-in' : 'custom', path: curr }]
+  }, [] as Hook[])
+}
+
+function parseHookName(name: string, isBuiltIn: boolean): [string, boolean] {
+  if (name.charAt(0) === '.') {
+    return [isBuiltIn ? name.substr(1) : calculateHash(name.substr(1)), false]
+  } else {
+    return [isBuiltIn ? name : calculateHash(name), true]
```

Comment:
What do you think of this approach? 

```js
const disabled= name.startsWith('.')
const name = disabled ? name.substr(1) : name

return { name: isBuiltIn ? name : calculateHash(name), disabled }
```

Reply:
5682deac554a265d4ccbd2ddcc84f6d897f5ba95